server.cc

```diff
@@ -108,58 +111,70 @@ class GRPCServiceImplementation final : public nvidia::inferenceserver::GRPCServ
           << nanoseconds.count() << std::endl;
   }
 
-  grpc::Status Status(
-		     ServerContext* context, 
-		     const StatusRequest* request, 
-		     StatusResponse* reply
-		     ) override {
+  grpc::Status ModelConfig(ServerContext* context,
+                           const ModelConfigRequest* request,
+                           ModelConfigResponse* reply) override {
+    //std::cout<<"In Config"<<std::endl;
+    if (request->name()=="facile_all_v2") {
+      auto config = reply->mutable_config();
+      config->set_name(request->name());
+      config->set_max_batch_size(16384);
+      return grpc::Status::OK;
+    }
+    else {
+      return grpc::Status::CANCELLED;
+    }
+  }
 
-    auto server_status = reply->mutable_server_status();
-    server_status->set_id("inference:0");
-    auto& model_status  = *server_status->mutable_model_status();
-    auto config = model_status["facile"].mutable_config();
-    config->set_max_batch_size(1600000);
-    config->set_name("facile");
-    auto input = config->add_input();
-    input->set_name("input");
-    input->set_data_type(DataType::TYPE_UINT16);
-    input->add_dims(32);
-    auto output = config->add_output();
-    output->set_name("output/BiasAdd");
-    output->set_data_type(DataType::TYPE_UINT16);
-    output->add_dims(1);
-    reply->mutable_request_status()->set_code(RequestStatusCode::SUCCESS);
-    nvidia::inferenceserver::RequestStatus request_status = reply->request_status();
-    nvidia::inferenceserver::ServerStatus  check_server_status  = reply->server_status();
-    return grpc::Status::OK;
+  grpc::Status ModelMetadata(ServerContext* context,
+                           const ModelMetadataRequest* request,
+                           ModelMetadataResponse* reply) override {
+    //std::cout<<"In Metadata"<<std::endl;
+    if (request->name()=="facile_all_v2") {
+      reply->set_name(request->name());
+      reply->add_versions(request->version());
+      auto input = reply->add_inputs();
+      input->set_name("input_DataConverter:FloatApFixed16F6Converter"); //designed to allowpassing of desired data converter
```

Comment:
the original first part of the input name should be kept, rather than using the generic "input" (& similar for output)

Reply:
Sorry, I don't understand what you mean here. Can you elaborate?

Reply:
`input->set_name("input_...")` should be `input->set_name(input->name()+"_...");`